plugins/header_rewrite/resources.cc

```diff
@@ -99,6 +104,24 @@ Resources::gather(const ResourceIDs ids, TSHttpHookID hook)
     }
     break;
 
+  case TS_HTTP_TXN_START_HOOK:
+    // Get TCP Info at transaction start
+    if (client_bufp && client_hdr_loc) {
```

Comment:
If I'm reading the code correctly, this condition would never be true because request header has not been read on `TS_HTTP_TXN_START_HOOK` yet.
https://docs.trafficserver.apache.org/en/9.0.x/developer-guide/plugins/hooks-and-transactions/index.en.html

Reply:
Actually, the pointers are available. We can't read request headers, but can add headers.